algo/uidlist.go

```diff
@@ -105,34 +105,45 @@ func IntersectCompressedWithBin(dec *codec.Decoder, q []uint64, o *[]uint64) {
 
 	// Pick the shorter list and do binary search
 	if ld < lq {
-		uids := dec.Uids()
-		for len(uids) > 0 {
-			for _, u := range uids {
-				qidx := sort.Search(len(q), func(idx int) bool {
-					return q[idx] >= u
-				})
-				if qidx >= len(q) {
-					return
-				}
-				if q[qidx] == u {
-					*o = append(*o, u)
-					qidx++
-				}
-				q = q[qidx:]
+		for {
+			block_uids := dec.Uids()
+			if len(block_uids) == 0 {
+				break
 			}
-			uids = dec.Next()
+			IntersectWithBin(block_uids, q, o)
+			last_uid := block_uids[len(block_uids)-1]
+			qidx := sort.Search(len(q), func(idx int) bool {
+				return q[idx] >= last_uid
+			})
+			if qidx >= len(q) {
+				return
+			}
+			q = q[qidx:]
```

Comment:
This is same as what you were doing earlier. The slice will keep increasing. I was hoping to avoid that.